src/pyodide/internal/snapshot.ts

```diff
@@ -264,8 +264,13 @@ function memorySnapshotDoImports(Module: Module): string[] {
         MetadataReader.getWorkerFiles('py')
       )
     );
-
-  const deduplicatedModules = [...new Set(importedModules)];
+  const moduleSet = new Set(importedModules);
+  // Importing pyodide before `finalizeBootstrap()` is called causes mayhem
+  // Make sure we don't do that!
+  // TODO: in Pyodide 0.27 we can call finalizeBootstrap() before taking the snapshot. Doing so will
+  // remove the need for this.
+  moduleSet.delete('pyodide');
```

Comment:
We already have some filtering logic in https://github.com/cloudflare/workerd/blob/07cb474032c16c09ca8750afcec07414b00dc44e/src/workerd/api/pyodide/pyodide.c%2B%2B#L351

Seems to only be filtering out `js` directly right now, but best to do the filtering in one place.